pkg/webui/console/containers/gateway-onboarding-form/index.js

```diff
@@ -84,39 +91,70 @@ const GatewayOnboardingForm = props => {
       const isUserOwner = _ownerId ? userId === _ownerId : true
       const ownerId = _ownerId ? _ownerId : userId
       const gatewayId = cleanValues.ids.gateway_id
+      setGatewayId(gatewayId)
 
       try {
         await dispatch(attachPromise(createGateway(ownerId, cleanValues, isUserOwner)))
         if (_create_api_key_cups) {
-          await generateCUPSApiKey(gatewayId)
+          const generatedCupsKey = await generateCUPSApiKey(gatewayId)
+          setCupsKey(generatedCupsKey.key)
         }
         if (_create_api_key_lns) {
-          await generateLNSApiKey(gatewayId)
+          const generatedLnsKey = await generateLNSApiKey(gatewayId)
+          const lbs_lns_secret = { value: btoa(generatedLnsKey.key) }
+          await dispatch(attachPromise(updateGateway(gatewayId, { lbs_lns_secret })))
+          setLnsKey(generatedLnsKey.key)
         }
-        onSuccess(gatewayId)
+        setModalVisible(_create_api_key_cups || _create_api_key_lns)
       } catch (error) {
         setError(error)
       }
     },
-    [dispatch, generateCUPSApiKey, generateLNSApiKey, onSuccess, userId],
+    [dispatch, generateCUPSApiKey, generateLNSApiKey, userId],
   )
 
   const handleSubmit = useCallback(
     (values, _, cleanValues) => handleRegistrationSubmit(values, cleanValues),
     [handleRegistrationSubmit],
   )
 
+  const downloadLns = useCallback(() => {
+    if (lnsKey) {
+      const content = composeDataUri(`Authorization: Bearer ${lnsKey}\r\n`)
+      downloadDataUriAsFile(content, 'tc.key')
+    }
+  }, [lnsKey])
+
+  const downloadCups = useCallback(() => {
+    if (cupsKey) {
+      const content = composeDataUri(`Authorization: Bearer ${cupsKey}\r\n`)
+      downloadDataUriAsFile(content, 'cups.key')
+    }
+  }, [cupsKey])
+
+  const modalApprove = useCallback(() => onSuccess(gatewayId), [onSuccess, gatewayId])
+
   return (
-    <Form
-      error={error}
-      onSubmit={handleSubmit}
-      initialValues={initialValues}
-      hiddenFields={['gateway_server_address', 'enforce_duty_cycle', 'schedule_anytime_delay']}
-      validationSchema={validationSchema}
-      validateAgainstCleanedValues
-    >
-      <GatewayProvisioningFormSection userId={userId} />
-    </Form>
+    <React.Fragment>
+      <GatewayApiKeysModal
```

Comment:
Cancel and I have download both send us to onSucess .

Reply:
Could you clarify this comment?